### PR TITLE
Suggestion query cleanup for "Review & Publish" button

### PR DESCRIPTION
**Pull request by samschaevitz**
fixes #

PR checklist:
- [n/a ] Update READ.me ?
- [n/a] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/react/Settings/utils/suggestions.ts [hunk 1]
```diff
@@ -1,24 +1,51 @@
...
+    [`_${name}`]: {
+      values: ['any'],
+    },
   };
   query.filters = filters;
   return query;
 }
+
+export function getReadyToReviewSuggestionsQuery(
+  matchingTemplateProperty: any,
```
Comment (bdittes) [inline app/react/Settings/utils/suggestions.ts]: Please type all these as PropertySchema (| undefined)

File: app/react/Thesauri/ThesaurusCockpit.tsx [hunk 2]
```diff
@@ -105,6 +110,24 @@ export class ThesaurusCockpitBase extends RouteHandler {
     );
   }
 
+  static async getAndPopulateSuggestionResults(
+    templates: { _id: string }[],
+    requestParams: RequestParams,
+    queryBuilderFunc: (prop: any, id: string) => any,
```
Comment (bdittes) [inline app/react/Thesauri/ThesaurusCockpit.tsx]: Please type all these as PropertySchema (| undefined)

File: app/react/Thesauri/ThesaurusCockpit.tsx [hunk 3]
```diff
@@ -120,24 +143,30 @@ export class ThesaurusCockpitBase extends RouteHandler {
     const assocProp = resolveTemplateProp(thesaurus, templates);
     thesaurus.property = assocProp;
 
-    // Get aggregate document count of documents with suggestions on this thesaurus
-    const allDocsWithSuggestions = await Promise.all(
-      templates.map((template: { _id: string }) => {
-        const reqParams = requestParams.set(getSuggestionsQuery(assocProp, template._id));
-        return api.search(reqParams);
-      })
+    // Query for documents with pending suggestions (published or not)
```
Comment (bdittes) [inline app/react/Thesauri/ThesaurusCockpit.tsx]: comment seems outdated

File: app/react/Thesauri/ThesaurusCockpit.tsx [hunk 4]
```diff
@@ -213,7 +244,8 @@ interface ThesaurusCockpitStore {
   thesauri: {
     thesaurus: IImmutable<ThesaurusSchema>;
     models: IImmutable<ClassifierModelSchema>[];
-    suggestions: IImmutable<SuggestionResultSchema>;
+    suggestionsTBPublished: IImmutable<SuggestionResultSchema>;
```
Comment (bdittes) [inline app/react/Thesauri/ThesaurusCockpit.tsx]: Do you need the full SuggestionResultSchema? Or is a number enough?

Generally, I'm not a fan of adding too many simple fields to the reducers - how about thesauri/suggestionStats = { TBPublished: number, TBReviewed: number}?

**Comment by samschaevitz** [comment]
Now with no more overcounting of documents in ThesauriList page view (s/summing per-value counts/totalRows of filter query)

File: app/react/Settings/utils/suggestions.ts (hunk 1 above)
Comment (samschaevitz) [reply app/react/Settings/utils/suggestions.ts]: Done.

File: app/react/Thesauri/ThesaurusCockpit.tsx (hunk 2 above)
Comment (samschaevitz) [reply app/react/Thesauri/ThesaurusCockpit.tsx]: Done.

File: app/react/Thesauri/ThesaurusCockpit.tsx (hunk 3 above)
Comment (samschaevitz) [reply app/react/Thesauri/ThesaurusCockpit.tsx]: removed

File: app/react/Thesauri/ThesaurusCockpit.tsx (hunk 4 above)
Comment (samschaevitz) [reply app/react/Thesauri/ThesaurusCockpit.tsx]: For TBPublished, that makes sense and I shall make it so.

For TBReviewed, we need the per-thesaurus-value counts as well so it cannot become just a number, unfortunately.

Comment (samschaevitz) [reply app/react/Thesauri/ThesaurusCockpit.tsx]: Ah, I now read your comment differently than I did at first. You're suggesting that, in addition to the per-value suggestions to be reviewed (fka 'suggestions'), I add another one that is 'suggestionsStats' and just have the summary there?

**Pull request merged**